@xen-orchestra/web-core/lib/components/PowerStateIcon.vue

```diff
@@ -0,0 +1,43 @@
+<template>
+  <UiIcon :class="className" :icon="icon" class="power-state-icon" />
+</template>
+
+<script lang="ts" setup>
+import UiIcon from '@core/components/ui/icon/UiIcon.vue'
+import { VM_POWER_STATE } from '@core/utils/xen-api/xen-api.enums'
```

Comment:
I don't think Web Core should have any notion of Xen Api.

Core components should be decoupled from the data as much as possible.